packages/iov-bns/src/encode.spec.ts

```diff
@@ -105,12 +107,26 @@ describe("Encode", () => {
         kind: TransactionKind.RegisterBlockchain,
         chainId: "registry-chain" as ChainId,
         signer: defaultSigner,
-        blockchainId: "wonderland" as ChainId,
+        blockchainId: toUtf8("wonderland") as BnsBlockchainId,
+        chain: {
+          chainId: "wonderland" as ChainId,
+          networkId: "7rg047g4h",
+          production: false,
+          enabled: true,
+          mainTickerId: "WONDER" as TokenTicker,
+          name: "Wonderland",
+        },
         codecName: "rules_of_wonderland",
         codecConfig: `{ rules: ["make peace not war"] }`,
       };
       const msg = buildMsg(registerBlockchain).issueBlockchainNftMsg!;
       expect(msg.id).toEqual(toAscii("wonderland"));
+      expect(msg.details!.chain!.chainID).toEqual("wonderland");
+      expect(msg.details!.chain!.networkID).toEqual("7rg047g4h");
```

Comment:
In the decode you ensure `chain` is present.
Here it looks like we are unsure of it's existence in the types.

I don't know what is correct, but should be consistent.

Reply:
>  Here it looks like we are unsure of it's existence in the types.

`msg` has the protobuf interface type. `details` and `chain` is coming as optional from the protobuf types in TypeScript. Since they are marked as non-null in
* https://github.com/iov-one/weave/blob/master/x/nft/blockchain/codec.proto#L41
* https://github.com/iov-one/weave/blob/master/x/nft/blockchain/codec.proto#L15
I assume both fields need to exist. Thus I force-cast them to non-null, which means I do the same as in decode.